### ISSUE TITLE
Pass parameters to build tooling

### ISSUE THREAD

**Issue by xeruf** [flutter/flutter#115752]
## Steps to Reproduce

Build https://aur.archlinux.org/packages/invoiceninja-desktop-git (or https://github.com/invoiceninja/admin-portal directly) with `flutter run`:

<img width="827" alt="image" src="https://user-images.githubusercontent.com/13354331/203054866-af60d579-9ffd-4280-9691-32ab9bbde7f5.png">

- I would like to get rid of the make warning, how can I pass `-Wno-dev` to make?
- Can I suppress the root warning? Since this is fakeroot, it does not matter.

**Comment by exaby73** [comment]
Hello @xeruf. Thank you for filing this issue. Can you please provide the following information?

- A [complete and minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) with clear steps to reproduce the issue. The sample should not include any 3rd party libraries. If this is an issue with InvoiceNinja only, a better place would be to raise an issue with them. 
- The output of `flutter doctor -v`

Please provide the information in the form of text. See how to make collapsible sections with Markdown [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections).

**Comment by xeruf** [comment]
I have no experience with flutter development and it should be unrelated to the actual code,
especially the root warning.

Here is the doctor:
```sh
❯ flutter doctor -v
[✓] Flutter (Channel stable, 3.3.8, on Arch Linux 6.0.9-arch1-1, locale en_IE.UTF-8)
    • Flutter version 3.3.8 on channel stable at /opt/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 52b3dc25f6 (13 days ago), 2022-11-09 12:09:26 +0800
    • Engine revision 857bd6b74c
    • Dart version 2.18.4
    • DevTools version 2.15.0

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /opt/android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/linux#android-setup for more details.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Linux toolchain - develop for Linux desktop
    • clang version 14.0.6
    • cmake version 3.25.0
    • ninja version 1.11.1
    • pkg-config version 1.8.0

[✓] Android Studio
    • Android Studio at /opt/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version openjdk version "11.0.13" 2021-10-19

[✓] Connected device (1 available)
    • Linux (desktop) • linux • linux-x64 • Arch Linux 6.0.9-arch1-1

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.

```

**Comment by exaby73** [comment]
The [complete and minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) I mentioned doesn't need to be code, but rather steps to reproduce this issue. I would need such a sample to verify the issue, else we cannot move forward with it.

**Comment by xeruf** [comment]
Steps:

1. `sudo flutter`

**Comment by exaby73** [comment]
Using `sudo` is not recommended with Flutter. And Flutter will give you a warning due to this. I need a way to reproduce this issue without any 3rd party dependencies. This includes the fakeroot environment as that is not something Flutter controls. Please provide a way to reliably reproduce this issue, with minimal code samples and steps to reproduce, else I won't be able to take this issue further

**Comment by xeruf** [comment]
But you don't need to reproduce the fakeroot environment.
The matter of fact is that `flutter` is run in an environment where it acts as root without any security implications, thus there should be a flag to disable this warning.

**Comment by huycozy** [comment]
@xeruf Can you share the source of `-Wno-dev` parameter? Is it [`invoiceninja`](https://github.com/invoiceninja/invoiceninja) parameter?

**Comment by exaby73** [comment]
> The matter of fact is that flutter is run in an environment where it acts as root without any security implications, thus there should be a flag to disable this warning.

@xeruf The sudo warning is different from the warning coming from `cmake`.  When running Flutter as sudo, you get the following warning:

```console
Woah! You appear to be trying to run flutter as root.
We strongly recommend running the flutter tool without superuser privileges.
```

This is different from the CMake one you wish to get rid of. Now speaking of this CMake warning, I can not reproduce this with a default app, even after running `sudo flutter build linux`. This warning appears to be originating from how InvoiceNinja has set up their build dependencies. 

Now while you request to pass parameters to build tooling may be valid, without a way to reproduce this error consistently, how would we know that this issue is solved? We cannot rely on InvoiceNinja's repository unfortunately. We need a small sample that we can test against this warning, to both verify it, and to test a potential fix against it if it comes to that. What I am asking of you, is to provide such a sample so that this issue can be taken further.

If you are unable to provide a sample, this issue while may be valid, is unactionable and therefore, would make it invalid.

**Comment by xeruf** [comment]
> This is different from the CMake one you wish to get rid of.

Since getting rid of the cmake warning seems to require more work than it's worth for me,
I am solely concerned about the root warning now.

Is it so hard to get from my initial enumeration that I want to get rid of _both_?

**Comment by eliasyishak** [comment]
@xeruf i am also struggling to reproduce the cmake warning, but if i'm reading this correctly, are you now trying to just get rid of the warning that is printed out when running as root below
```
   Woah! You appear to be trying to run flutter as root.
   We strongly recommend running the flutter tool without superuser privileges.
  /
📎
```

**Comment by xeruf** [comment]
Yes, flag or environment variable or whatever, I don't care - I just don't want to always have a big warning popping up when flutter is upgraded on arch as package builds are *always* run in fakeroot there.
